src/datadog/trace_segment.cpp

```diff
@@ -179,12 +209,15 @@ void TraceSegment::span_finished() {
     }
   }
 
-  // Origin is repeated on all spans.
-  if (origin_) {
-    for (const auto& span_ptr : spans_) {
-      SpanData& span = *span_ptr;
+  // Some tags are repeated on all spans.
+  for (const auto& span_ptr : spans_) {
+    SpanData& span = *span_ptr;
+    if (origin_) {
       span.tags[tags::internal::origin] = *origin_;
     }
+    span.numeric_tags[tags::internal::process_id] = cached_process_id();
```

Comment:
I still have doubts about the tags / numeric tags separation for essentially random values.
IMO it'd be reasonable for things that could either be treated as counters, gauges or categories (ie: enumerated values) but a PID is none of those things.

Reply:
My guess is somebody thought "it's a number, put it in the number thing."

Also worth remembering that numeric tags are floating point, not that it matters here.